CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.21)
```

Comment:
why?

Reply:
I was trying out some features in newer cmake like CMakePresets.json, its not needed since I didn't add them eventually